src/track/beatmap.cpp

```diff
@@ -268,6 +274,93 @@ double BeatMap::findNthBeat(double dSamples, int n) const {
     return -1;
 }
 
+bool BeatMap::findPrevNextBeats(double dSamples,
```

Comment:
added a TODO.  We probably don't want to deal with iterators because of the locking issues.